src/main/scala/org/globalforestwatch/summarystats/forest_change_diagnostic/ForestChangeDiagnosticDataLossYearlyTwoCategory.scala

```diff
@@ -0,0 +1,82 @@
+package org.globalforestwatch.summarystats.forest_change_diagnostic
```

Comment:
Maybe this wouldn't be possible with how encoders work in Scala, but if I was to do this in Python, instead of making two classes for each number of classes I needed, I would just make one class that accepts a list of classes. Then when doing the encoding, figure out the nesting based on the number of classes.

Reply:
Same for the two-level one above for ApproxYear

Reply:
That's a good suggestion about trying to use of list of classes/categories.  But I think it would be very hard to get the encoding/decoding to work correctly, when we want to convert a list of categories to a nested set of categories during encoding to JSON and vice versa for decoding.  Note that there are actually multiple kinds of encoding/decoding going on, including circe JSON and also shapeless encoding for transferring objects in Spark, but some encoders/decoders are being created automatically.  So, rather than trying to write a bunch of explicit JSON and other encoding/decoding to do a general number of categories, I think I'll keep the code as is.